Bulk release notes (#6158)
dbt-labs/docs.getdbt.com: website/docs/docs/dbt-versions/release-notes.md

## September 2024

> 

Reviewer: non blocker but i wonder if we can make it explicit that these are changes for versionless users and upcoming 1.9? maybe its best to keep it simple but wanted to flag that!

e.g 

## September 2024
#### dbt Cloud Versionless (coming in 1.9 release)
- your bulk updates
#### Also available this month
rest of the updates for sept

> - **New**: You can set descriptions in your YAML files to document [singular data tests](/docs/build/data-tests#singular-tests).

Reviewer: is this link right? it's not linking to anything that's 'singular-tests' in the data-test page. is it meant to link to 'singular-data-tests'?

Reviewer 2: I think we had to roll this back - so we may want to hold off on adding this to the release notes

> - **New**: The snapshot config can now be set in schema yml file.

Reviewer: can we link to this config doc ?
```suggestion
- **New**: The snapshot configuration can now be set in the schema YAML file, making it easier to manage and organize your project’s configuration in one place. 
```

> - **New**: The snapshot configuration can now be set in the schema YAML file, making it easier to manage and organize your project’s configuration in one place. 

Reviewer 2: should we link to docs here?

> - **New**: You will now get a warning if the data type of the `updated_at` column does not match the adapter-configured default.

Reviewer 2: Let's specify that this is for snapshots and link to the docs?

> - **New**: You can now define the names of the columns created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields) using custom configurations.

Reviewer 2: ```suggestion
- **New**: You can now customize the names of the meta columns dbt creates in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields) using the new `snapshot_meta_column_names` configuration.
```

> - **New**: dbt `test` now enables passing values like `unit_tests`, `tests`, `--resource-type` and `--exclude-resource-type` for CLI flags and `DBT_RESOURCE_TYPES` and `DBT_EXCLUDE_RESOURCE_TYPES)` in the environment variables.

Reviewer: ```suggestion
- **New**: dbt `test` now enables passing values like `unit_tests`, `tests`, `--resource-type` and `--exclude-resource-type` for CLI flags and `DBT_RESOURCE_TYPES` and `DBT_EXCLUDE_RESOURCE_TYPES)`  for environment variables.
```

> - **New**: dbt `test` now enables passing values like `unit_tests`, `tests`, `--resource-type` and `--exclude-resource-type` for CLI flags and `DBT_RESOURCE_TYPES` and `DBT_EXCLUDE_RESOURCE_TYPES)`  for environment variables.

Reviewer 2: I don't follow this... maybe:

> dbt `test` now enables excluding/including specific resource types using the `--resource-type` and `--exclude-resource-type` CLI flags and the `DBT_RESOURCE_TYPES` and `DBT_EXCLUDE_RESOURCE_TYPES)` environment variables.

> - **Enhancement**: When [defining constraints](/reference/resource-properties/constraints#defining-constraints) you can use the improved syntax to define foreign keys, which uses `ref`. The recommended syntax means dbt automatically resolves references to other models, ensuring they capture dependencies and work across different environments.

Reviewer 2: ```suggestion
- **Enhancement**: When [defining constraints](/reference/resource-properties/constraints#defining-constraints) you can now define foreign keys using `ref`, so that dbt will automatically resolves references to other models, ensuring they capture dependencies and work across different environments.
```

@jtcohen6 lmk if this looks good to you
